modules/react/menu/lib/MenuItem.tsx

```diff
@@ -148,39 +166,67 @@ export const StyledMenuItem = createComponent('button')({
   },
 });
 
-export const useMenuItem = composeHooks(
-  createElemPropsHook(useMenuModel)(
-    (model, ref, elemProps: {'data-id': string} = {'data-id': ''}) => {
-      const {localRef, elementRef} = useLocalRef(ref as React.Ref<HTMLElement>);
-      const id = elemProps['data-id'];
-
-      // focus on the item with the cursor
-      React.useLayoutEffect(() => {
-        if (model.state.mode === 'single') {
-          if (model.state.cursorId === id) {
-            // delay focus changes to allow PopperJS to position
-            requestAnimationFrame(() => {
-              localRef.current?.focus();
-            });
-          }
+export const useMenuItemArrowReturn = createElemPropsHook(useMenuModel)(model => {
+  return {
+    onKeyDown(event: React.KeyboardEvent) {
+      const styles = getComputedStyle(event.currentTarget);
+      if (event.key === 'ArrowLeft' && styles.direction === 'ltr' && model.UNSTABLE_parentModel) {
```

Comment:
nice, this is what you mean by events based on directionality instead of the isRTL/?

Reply:
Yes

Reply:
I think a utility function in `common` would be nice:

```js
if (event.key === 'ArrowLeft' && isRTL(event.current) && model.UNSTABLE_parentModel) {
```

where `isRTL` takes in an element and returns `true` if the element's direction is RTL